backend/src/data/repositories/course/course.repository.ts

```diff
@@ -17,15 +17,27 @@ class Course {
 
   public getAll(filteringOpts: {
     categoryId: number | null;
+    title: string;
   }): Promise<CourseGetResponseDto[]> {
     const { categoryId } = filteringOpts ?? {};
+    const { title = '' } = filteringOpts ?? {};
```

Comment:
Could it be simplified to:
const { categoryId,  title = '' } = filteringOpts ?? {};